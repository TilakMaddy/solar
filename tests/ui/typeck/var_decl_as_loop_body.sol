contract C {
    function var_decl_inside_loops() external {
        for (uint256 i = 0; i < 100; ++i) {
            uint256 m_count = i + 1 * 2;
        }

        for (uint256 i = 0; i < 100; ++i) uint256 m_count = i + 1 * 2; //~ ERROR: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block

        for (uint256 i = 0; i < 100; ++i)
            for (uint256 j = 0; i < 100; ++j) {
                uint256 k = i + j;
            }

        for (uint256 i = 0; i < 100; ++i)
            for (uint256 j = 0; i < 100; ++j) uint256 k = i + j; //~ ERROR: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block

        while (true) uint256 x = 4; //~ ERROR: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block

        do uint256 x = 4; while (true); //~ ERROR: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block

        unchecked {
            {
                {
                    for (uint256 i = 0; i < 10; ++i) uint256 y = 0; //~ ERROR: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block
                }
            }
        }
    }
}
