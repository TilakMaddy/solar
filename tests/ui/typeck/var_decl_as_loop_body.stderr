error: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block
  --> ROOT/tests/ui/typeck/var_decl_as_loop_body.sol:LL:CC
   |
LL | ...   for (uint256 i = 0; i < 100; ++i) uint256 m_count = i + 1 * 2;
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |

error: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block
  --> ROOT/tests/ui/typeck/var_decl_as_loop_body.sol:LL:CC
   |
LL | ...   for (uint256 j = 0; i < 100; ++j) uint256 k = i + j;
   |                                         ^^^^^^^^^^^^^^^^^
   |

error: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block
  --> ROOT/tests/ui/typeck/var_decl_as_loop_body.sol:LL:CC
   |
LL | ...   while (true) uint256 x = 4;
   |                    ^^^^^^^^^^^^^
   |

error: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block
  --> ROOT/tests/ui/typeck/var_decl_as_loop_body.sol:LL:CC
   |
LL | ...   do uint256 x = 4; while (true);
   |          ^^^^^^^^^^^^^
   |

error: aborting due to 4 previous errors

